nomad/resource_job_v2.go

```diff
@@ -0,0 +1,1682 @@
+
+						paths := make([]*api.ConsulExposePath, 0)
+						for _, cep := range ec["path"].([]interface{}) {
+							p := cep.(map[string]interface{})
+
+							path := &api.ConsulExposePath{
+								Path:          p["path"].(string),
+								Protocol:      p["protocol"].(string),
+								LocalPathPort: p["local_path_port"].(int),
+								ListenerPort:  p["listener_port"].(string),
+							}
+
+							paths = append(paths, path)
+						}
+
+						exposeConfig = &api.ConsulExposeConfig{
+							Path: paths,
+						}
+					}
+
+					consulProxy = &api.ConsulProxy{
+						LocalServiceAddress: p["local_service_address"].(string),
+						LocalServicePort:    p["local_service_port"].(int),
+						ExposeConfig:        exposeConfig,
+						Upstreams:           upstreams,
+						Config:              config,
+					}
+				}
+
+				sidecarService = &api.ConsulSidecarService{
+					Tags:  tags,
+					Port:  ss["port"].(string),
+					Proxy: consulProxy,
+				}
+			}
+
+			connect = &api.ConsulConnect{
+				Native:         cn["native"].(bool),
+				SidecarService: sidecarService,
+				SidecarTask:    sidecarTask,
+			}
+		}
+
+		service := &api.Service{
+			Meta:              getMapOfString(s["meta"]),
+			Name:              s["name"].(string),
+			PortLabel:         s["port"].(string),
+			Tags:              tags,
+			CanaryTags:        canaryTags,
+			EnableTagOverride: s["enable_tag_override"].(bool),
+			AddressMode:       s["address_mode"].(string),
+			TaskName:          s["task"].(string),
+			Checks:            checks,
+			Connect:           connect,
+			CanaryMeta:        getMapOfString(s["canary_meta"]),
+		}
+
+		services = append(services, service)
+	}
+
+	return services, nil
+}
+
+func getLogConfig(d interface{}) *api.LogConfig {
+	for _, lg := range d.([]interface{}) {
+		l := lg.(map[string]interface{})
+
+		return &api.LogConfig{
+			MaxFiles:      getInt(l, "max_files"),
+			MaxFileSizeMB: getInt(l, "max_file_size"),
+		}
+	}
+
+	return nil
+}
+
+func getResources(d interface{}) *api.Resources {
+	for _, rs := range d.([]interface{}) {
+		r := rs.(map[string]interface{})
+
+		devices := make([]*api.RequestedDevice, 0)
+		for _, dv := range r["device"].([]interface{}) {
+			d := dv.(map[string]interface{})
+
+			count := uint64(d["count"].(int))
+			device := &api.RequestedDevice{
+				Name:        d["name"].(string),
+				Count:       &count,
+				Constraints: getConstraints(d["constraint"]),
+				Affinities:  getAffinities(d["affinity"]),
+			}
+
+			devices = append(devices, device)
+		}
+
+		return &api.Resources{
+			CPU:      getInt(r, "cpu"),
+			MemoryMB: getInt(r, "memory"),
+			Networks: getNetworks(r["network"]),
+			Devices:  devices,
+		}
+	}
+
+	return nil
+}
+
+func getVault(d interface{}) *api.Vault {
+	for _, vlt := range d.([]interface{}) {
+		v := vlt.(map[string]interface{})
+
+		policies := make([]string, 0)
+		for _, p := range v["policies"].([]interface{}) {
+			policies = append(policies, p.(string))
+		}
```

Comment:
Would it make sense to encapsulate the logic f parsing a `[]interface{}` as `[]string` into a function, like `getMapOfString`? This seems to be used in a lot of places.